public/app/plugins/datasource/prometheus/datasource.test.ts

```diff
@@ -97,6 +98,72 @@ describe('PrometheusDatasource', () => {
         expect(response[0].state).toBe(LoadingState.Done);
       });
     });
+
+    describe('step parameter', () => {
+      const to = dateTime();
+      const from = dateTime(to).subtract(6, 'hours');
+      const sixHoursRange = { from, to, raw: { from: 'now-6h', to: 'now' } };
+
+      describe('When set to exact', () => {
+        it('should return an error if interval is modified', async () => {
+          const request = createDataRequest([{ interval: '1s', stepMode: 'exact' }], {
+            range: sixHoursRange,
+          });
+
+          await expect(ds.query({ ...request, app: CoreApp.Dashboard })).toEmitValuesWith((response) => {
+            expect(response[0].error).toBeDefined();
+          });
+
+          await expect(ds.query({ ...request, app: CoreApp.Explore })).toEmitValuesWith((response) => {
```

Comment:
Out of curiosity: Why asserting twice for the same thing? If it's to check the error is displayed each time - shouldn't we run one query without an error in between to ensure it's not a stale error message from the previous query? Or assert that the message has changed instead?

Reply:
it checks if the error is present or not both in explore and in dashboards.
when running queries different methods are called for explore and dashboards, so I'm testing both scenarios

Reply:
🤦 didn't notice CoreApp.Explore / Dashboard